src/main/core/support/configuration.rs

```diff
@@ -953,6 +953,89 @@ impl<'de> serde::Deserialize<'de> for ProcessArgs {
     }
 }
 
+#[derive(Debug, Copy, Clone)]
+pub struct Signal(nix::sys::signal::Signal);
+
+impl serde::Serialize for Signal {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.0.as_str())
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for Signal {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        struct SignalVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for SignalVisitor {
+            type Value = Signal;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a signal string or integer")
+            }
+
+            fn visit_string<E>(self, v: String) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                self.visit_str(&v)
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                nix::sys::signal::Signal::from_str(v)
+                    .map(Signal)
+                    .map_err(|_e| E::custom(format!("Invalid signal string: {v}")))
```

Comment:
Maybe add to the documentation (and maybe the doc comment / cli help) what format this needs to be in. For example, uppercase or lowercase, the shortened form (ex: `TERM`) or long form (`SIGTERM`), etc.

Reply:
Done